param/__init__.py

```diff
@@ -1630,3 +1630,29 @@ def _checkBounds(self, val):
                 if too_low or too_high:
                     raise ValueError("Parameter '%s' %s bound must be in range %s"
                                      % (self._attrib_name, bound, self.rangestr()))
+
+
+class DateRange(Range):
+    """
+    A date range specified as (start_date, end_date).
+
+    Dates must be specified as datetime-like types (see
+    param.dt_types).
```

Comment:
I'm going to deal with naming/doc of date vs datetime in a separately (because applies to existing Date parameter also).